### PR TITLE
MGMT-3026 - invoke fio from installcmd

### PR DESCRIPTION
**Pull request by danielerez**
- Invoke fio_perf_check agent command as part of installcmd flow.
- Handled error in inventory (handleReplyError) by looking for a specific fio exit code (FioPerfCheckCmdExitCode).

### CODE REVIEW

File: internal/host/fioperfcheckcmd.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	FioPerfCheckCmdExitCode int64 = 126
```
Comment (filanov) [inline internal/host/fioperfcheckcmd.go:15]: podman have 126 maybe something really big just to make sure that it doesn't conflict with any other error? 
https://github.com/containers/podman/issues/367

File: internal/host/installcmd_test.go [hunk 2]
```diff
@@ -29,12 +29,13 @@ var DefaultInstructionConfig = InstructionConfig{
...
 	ControllerImage:     "quay.io/ocpmetal/assisted-installer-controller:latest",
 	InventoryImage:      "quay.io/ocpmetal/assisted-installer-agent:latest",
+	FioPerfCheckImage:   "quay.io/ocpmetal/assisted-installer-agent:latest",
 	InstallationTimeout: 120,
 	ReleaseImage:        "quay.io/openshift-release-dev/ocp-release@sha256:eab93b4591699a5a4ff50ad3517892653f04fb840127895bb3609b3cc68f98f3",
 	ReleaseImageMirror:  "local.registry:5000/ocp@sha256:eab93b4591699a5a4ff50ad3517892653f04fb840127895bb3609b3cc68f98f3",
 }
 
-var _ = Describe("installcmd", func() {
+var _ = FDescribe("installcmd", func() {
```
Comment (filanov) [inline internal/host/installcmd_test.go:38]: What is `FDescribe`?

**Review by filanov — COMMENTED** [review]
Maybe you should add subsystem test that check this specific error flow. and get the events and the status info of the host to make sure that everything is set correctly.

File: internal/host/installcmd_test.go [hunk 3]
```diff
@@ -350,8 +351,15 @@ func validateInstallCommand(reply *models.Step, role models.HostRole, clusterId
 		"--boot-device /dev/sdb --host-id %s --openshift-version 4.5 --mco-image mcoImage " +
 		"--controller-image %s --url %s --insecure=false --agent-image %s --installation-timeout %s"
 
+	fioPerfCheckCmd := "podman run --privileged --net=host --rm --quiet -v /dev:/dev:rw -v /var/log:/var/log " +
+		"-v /run/systemd/journal/socket:/run/systemd/journal/socket " +
+		"quay.io/ocpmetal/assisted-installer-agent:latest fio_perf_check " +
```
Comment (rollandf) [inline internal/host/installcmd_test.go:356]: "quay.io/ocpmetal/assisted-installer-agent:latest" should not be hard coded

File: internal/host/installcmd_test.go (hunk 2 above)
Comment (rollandf) [reply internal/host/installcmd_test.go:38]: Focus for unit-test, need to remove

File: internal/host/instructionmanager.go [hunk 4]
```diff
@@ -56,6 +56,7 @@ type InstructionConfig struct {
 	DhcpLeaseAllocatorImage      string `envconfig:"DHCP_LEASE_ALLOCATOR_IMAGE" default:"quay.io/ocpmetal/assisted-installer-agent:latest"`
 	APIVIPConnectivityCheckImage string `envconfig:"API_VIP_CONNECTIVITY_CHECK_IMAGE" default:"quay.io/ocpmetal/assisted-installer-agent:latest"`
```
Comment (rollandf) [inline internal/host/instructionmanager.go:57]: need to add to openshift/template.yaml

Comment (rollandf) [reply internal/host/instructionmanager.go:58]: Done

File: internal/host/fioperfcheckcmd.go (hunk 1 above)
Comment (rollandf) [reply internal/host/fioperfcheckcmd.go]: Changed to 222

File: internal/host/installcmd_test.go (hunk 2 above)
Comment (rollandf) [reply internal/host/installcmd_test.go]: Removed

File: internal/host/host.go [hunk 5]
```diff
@@ -240,6 +241,20 @@ func (m *Manager) populateDisksEligibility(inventoryString string) (string, erro
 	return string(result), nil
 }
 
+func (m *Manager) HandlePrepareInstallationFailure(ctx context.Context, h *models.Host, reason string) error {
```
Comment (filanov) [inline internal/host/host.go:244]: isn't it the same as `HandleInstallationFailure`?

Comment (rollandf) [reply internal/host/host.go:244]: It adds the reason string.

**Review by filanov — APPROVED** [review]

**Pull request merged**